salt/_modules/metalk8s.py

```diff
@@ -287,3 +288,40 @@ def check_pillar_keys(keys, refresh=True, pillar=None, raise_error=True):
             return False
 
     return True
+
+
+def format_slots(data):
+    """Helper to replace slots in nested dictionnary
+
+    "__slots__:salt:module.function(arg1, arg2, kwarg1=abc, kwargs2=cde)
+
+    Arguments:
+        data: Data structure to format
+    """
+    slots_callers = {
+        "salt": __salt__
+    }
+
+    if isinstance(data, list):
+        return [format_slots(elt) for elt in data]
+
+    if isinstance(data, dict):
+        return {key: format_slots(value) for key, value in data.items()}
+
+    if isinstance(data, basestring) and data.startswith('__slot__:'):
+        fmt = data.split(":", 2)
+        if len(fmt) != 3:
+            log.warning("Malformed slot: %s", data)
+            return data
+        if fmt[1] not in slots_callers:
+            log.warning(
+                "Malformed slot, only %s supported: %s",
+                slots_callers.keys(), data
```

Comment:
The ordering of values makes things weird to read IMHO:

```suggestion
                "Malformed slot '%s': invalid caller, must use one of '%s'",
                data, "', '".join(slots_callers.keys())
```